### DISCUSSION TITLE
Cannot read base64 strings of image/x-icon MIME type

### DISCUSSION THREAD

**Discussion by erolo861** [ImageMagick/ImageMagick#2517] [Help]
I originally had this problem trying to read a ICO image from a base64 string in Imagick, but I later verified that the same exact problem also happens using the command line. I'm using version ImageMagick-7.0.10-29-portable-Q16-x86.

The command I'm using is:
`magick identify "inline:data:image/png;base64,iVBORw0KGgoAAAA.....(rest of base64 string)"`

The above command reads the base64 string of a PNG image, and works as intended, returning
`PNG 32x32 32x32+0+0 8-bit sRGB 428B 0.000u 0:00.000`.

However, when I use the base64 string of a ICO image, such as
`magick identify "inline:data:image/x-icon;base64,AAABAAMAMDAQ.....(rest of base64 string)"`,
it returns
`identify: no decode delegate for this image format `' @ error/blob.c/BlobToImage/458.`.

The string is correct, I've verified with this online converter https://base64.guru/converter/decode/image.
Here's the complete command if anyone wants to try it: https://justpaste.it/95icv

The list of supported formats returned using 'identify -list formats also includes ICO and ICON.

Please help me understand the problem.

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT master branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://www.imagemagick.org/download/beta/ by sometime tomorrow.
